spark-tensorflow/trainer/preprocess_artifacts_local.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#            http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import shutil
+import csv
+
+
+
+# TODO: would be better to figure out how to have Scala output it correctly
```

Comment:
The main issue is we want to write to GCS if there's ` gs` prefix so that we don't require a local computer to run it all on the Cloud. And it seemed easiest to save/load as dataframe. Since dataframe save already will know how to correctly save to GCS, it's *almost* what we want if we could just pick a filename, yes the alternative is to write code to save/load it locally or on GCS manually. It's not a ton of code but seemed like it would be unnecessary if DataFrame supported one extra feature. But yes should really get rid of the Python scripts anyway.